### PR TITLE
[DOCS] Document range enrich policy

### PR DESCRIPTION
**Pull request by mjmbischoff**
Adding the docs similar to the match policy for the new range policy

### CODE REVIEW

**Review by jrodewig — COMMENTED** [review]
Thanks @mjmbischoff.

I pushed some commits to update the API docs, remove some unneeded snippet comments, publish your example, and clean up some of my own mistakes.

I left some suggestions and comments. My only major outstanding question is what happens if the processor's `field` contains an array? The docs aren't clear right now.

**Review by jrodewig — APPROVED** [review]
Thanks for clarifying @mjmbischoff. I pushed some additional commits to clean up some of my additions based on your comment and to move the `include` statements to the right place.

If those changes look okay to you, this LGTM. Since I contributed, I'd like to get final approval from @martijnvg before merging.

**Review by martijnvg — APPROVED** [review]
Thanks @mjmbischoff and @jrodewig! 
This LGTM

**Pull request merged**
